content/tutorials/channel-occupancy-events.textile

```diff
@@ -91,114 +80,170 @@ Since we'll be using JavaScript in this tutorial, the best way to display the re
+  console.log('adding subscriber instance');
+  regularChannel.subscribe('test-data', (data) => {
+    //do whatever
+    console.log('Subscription working');
+  });
 }
 
-function enterPresence() {
-    resultArea.value += ('\n[LOCAL LOG - ' + (new Date().toLocaleTimeString()) + ' ]: Entering presence\n')
-    var ably = new Ably.Realtime({
-        key: apiKey
-    });
-    var regularChannel = ably.channels.get("regular-channel")
-    regularChannel.presence.enter();
+function addPublisherInstanceWithPresence() {
+  resultArea.value +=
+    '\n[LOCAL LOG - ' +
+    new Date().toLocaleTimeString() +
+    ' ]: Adding new publisher instance\n';
+  var myId = 'clientId-' + Math.random().toString(36).substr(2, 16);
+  var ably = new Ably.Realtime({
+    key: apiKey,
+    clientId: myId,
+  });
+  var regularChannel = ably.channels.get(regularChannelName);
+  console.log('adding publisher instance');
+  regularChannel.publish('test-data', {
+    data: 'Dummy Data',
+    time: Date.now(),
+  });
+  regularChannel.presence.enter();
 }
 
-function leavePresence() {
-    resultArea.value += ('\n[LOCAL LOG - ' + (new Date().toLocaleTimeString()) + ' ]: Leaving presence\n')
-    var ably = new Ably.Realtime({
-        key: apiKey
-    });
-    var regularChannel = ably.channels.get("regular-channel")
-    regularChannel.presence.leave();
+function addSubscriberInstanceWithPresence() {
+  resultArea.value +=
+    '\n[LOCAL LOG - ' +
+    new Date().toLocaleTimeString() +
+    ' ]: Adding new subscriber instance\n';
+  var myId = 'clientId-' + Math.random().toString(36).substr(2, 16);
+  var ably = new Ably.Realtime({
+    key: apiKey,
+    clientId: myId,
+  });
+  var regularChannel = ably.channels.get(regularChannelName);
+  console.log('adding subscriber instance');
+  regularChannel.subscribe('test-data', (data) => {
+    //do whatever
+    console.log('Subscription working');
+  });
+  regularChannel.presence.enter();
 }
 ```
 
+You are now ready to test your code. You can run the code locally by loading @index.html@ directly into your web browser. Then click the buttons provided to add occupants. Observe that the occupancy metrics update accordingly.
+
+The following provides a basic explanation of the occupancy metrics:
+
+- connections := the number of connections to the channel.
+- publishers := the number of connections attached to the channel that are authorized to publish.
+- subscribers := the number of connections attached that are authorized to subscribe to messages.
+- presenceSubscribers := the number of connections that are authorized to subscribe to presence messages.
+- presenceConnections := the number of connections that are authorized to enter members into the presence channel.
+- presenceMembers := the number of members currently entered into the presence channel.
+
 "See this step in Github":https://github.com/ably/tutorials/commit/occupancy-javascript-step5
 
 h2(#live-demo). Step 6 - Live Demo
```

Comment:
The live demo seems to be broken. Not looked into why

Reply:
This is because {{API_KEY}} generates a key that has a detached application. Web team are looking into it. The work-around would be to use a hard-coded key.